chia/cmds/rpc.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+import json
+import sys
+from typing import Any, Dict, List, Optional, TextIO
+
+import click
+from aiohttp import ClientResponseError
+
+from chia.util.config import load_config
+from chia.util.default_root import DEFAULT_ROOT_PATH
+from chia.util.ints import uint16
+
+services: List[str] = ["farmer", "full_node", "harvester", "wallet"]
+
+
+async def call_endpoint(
+    service: str, host: str, port: uint16, endpoint: str, request: Dict[str, Any], config: Dict[str, Any]
+) -> Dict[str, Any]:
+    from chia.rpc.rpc_client import RpcClient
+
+    try:
+        client = await RpcClient.create(host, port, DEFAULT_ROOT_PATH, config)
+    except Exception as e:
+        raise Exception(f"Failed to create RPC client {service}: {e}")
+    result: Dict[str, Any]
+    try:
+        result = await client.fetch(endpoint, request)
+    except ClientResponseError as e:
+        if e.code == 404:
+            raise Exception(f"Invalid endpoint for {service}: {endpoint}")
+        raise e
+    except Exception as e:
+        raise Exception(f"Request failed: {e}")
+    finally:
+        client.close()
+        await client.await_closed()
+    return result
+
+
+def print_result(json_dict: Dict[str, Any]) -> None:
+    print(json.dumps(json_dict, indent=4, sort_keys=True))
```

Comment:
This might lead to religeous war, but I'd suggest `indent=2` for it does not consume console width.

Reply:
Fewer indent reduces amount of movement of eyeball and it saves my eye health.
But I won't justify my indent preference any more...

![image](https://user-images.githubusercontent.com/84098616/158591131-e8419442-1f9b-4cfa-9aa8-62a1b851ca42.png)

![image](https://user-images.githubusercontent.com/84098616/158591168-a340fb09-780d-4ee6-845d-46dace420020.png)